### PR TITLE
[SGD] RaySGD v2 skeleton code

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

Basic skeleton and code structure for RaySGD v2. This should give a birds-eye view of how the functionality is organized.

- `trainer.py`
  - `Trainer`: Main class for performing distributed training that the user interfaces with. 
  -  `WorkerConfig`: Config class that the user can populate with configurations for starting distributed workers (number of actors, resources, etc.).
- `backend.py`
  - `BackendExecutor`: Abstract class for executing code in a distributed manner.  
  - `BackendConfig`: Config class that the user can populate with configurations for their backend executor. Individual implementations will subclass this with additional attributes.
- `actor_group.py`
  - `ActorGroup`: Generic utility class for handling a group of actors, including setting them up, executing code on them, and shutting them down.
  - `DistributedActor`: A single actor that is part of an `ActorGroup`.
- `backends/torch.py`: Specific implementation for executing distributed PyTorch.

## Related issue number

n/a

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/sgd/v2/actor_group.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ActorGroup:
+    pass
+
+
+class DistributedActor:
```
Comment (richardliaw) [inline python/ray/util/sgd/v2/actor_group.py:5]: What does DistributedActor do? Is this more like 'Worker'?

File: python/ray/util/sgd/v2/backend.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class BackendExecutor:
```
Comment (richardliaw) [inline python/ray/util/sgd/v2/backend.py:1]: should this be in backends/?

File: python/ray/util/sgd/v2/actor_group.py (hunk 1 above)
Comment (amogkam) [reply python/ray/util/sgd/v2/actor_group.py:5]: would it be better to rename this to `BaseActor`?

File: python/ray/util/sgd/v2/backend.py (hunk 2 above)
Comment (amogkam) [reply python/ray/util/sgd/v2/backend.py:1]: +1

File: python/ray/util/sgd/v2/backends/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from ray.util.sgd.v2.backends.torch import TorchConfig, TorchExecutor
```
Comment (amogkam) [inline python/ray/util/sgd/v2/backends/__init__.py:1]: I think this file can actually just be empty

File: python/ray/util/sgd/v2/actor_group.py (hunk 1 above)
Comment (matthewdeng) [reply python/ray/util/sgd/v2/actor_group.py:5]: Will rename to `WorkerGroup` and `BaseWorker`.

**Review by amogkam — APPROVED** [review]

**Pull request merged**